File: 프로그래머스/0/181934. 조건 문자열/조건 문자열.java
class Solution {
    public boolean nIsGreaterThanM (String ineq) {
        if(ineq.equals(">")) return true;
        return false;
    }
    
    public boolean nIsSameWithM (String eq) {
        if(eq.equals("=")) return true;
        return false;
    }
    
    public int solution(String ineq, String eq, int n, int m) {
        boolean val = false;
        if(nIsGreaterThanM(ineq)){
            if(nIsSameWithM(eq) && n >= m) val = true;
            else if(!nIsSameWithM(eq) && n > m) val = true;
        }else{
            if(nIsSameWithM(eq) && n <= m) val = true;
            else if(!nIsSameWithM(eq) && n < m ) val = true;
        }
        return val ? 1 : 0;
    }
}

File: 프로그래머스/0/181934. 조건 문자열/조건 문자열.js
function solution(ineq, eq, n, m) {
    return calc(typeClassy(ineq, eq), n, m) ? 1 : 0
}

const typeClassy = (ineq, eq) => {
    const ret = ineq === ">" ? eq === "=" ? 0 : 1 : eq === "=" ? 2 : 3
    return ret
}

const calc = (tc, n, m) => {
    switch(tc){
      case 0: return n >= m 
      case 1: return n > m  
      case 2: return n <= m 
      case 3: return n < m        
    }
}

